aries_cloudagent/wallet/models/wallet_record.py

```diff
@@ -106,6 +106,12 @@ def requires_external_key(self) -> bool:
         else:
             return True
 
+    def update_settings(self, settings: dict):
+        """Update settings."""
+        if "wallet.id" in settings:
+            del settings["wallet.id"]
```

Comment:
Maybe if `wallet.id` is present it should throw an error instead of deleting the property? I think this will help with preventing unexpected behaviour.

Reply:
Thanks for your suggestion. It is applied.